src/sageworks/utils/extract_model_artifact.py

```diff
@@ -70,7 +74,74 @@ def get_model_data_uri(self) -> str:
             raise ValueError("ModelDataUrl not found in the model package description")
 
     @staticmethod
-    def download_and_extract_model(model_artifact_uri):
+    def load_from_json(tmpdir):
+        # Find the model file in the extracted directory
+        model_files = glob.glob(os.path.join(tmpdir, "*_model.json"))
+        if not model_files:
+            return None
+
+        # Instantiate return model
+        model_return = None
+
+        # Check each model_file for an XGBModel object
+        for model_file in model_files:
+            # Get json and get model type
+            with open(model_file, "rb") as f:
+                model_json = json.load(f)
+            model_type = json.loads(model_json.get("learner").get("attributes").get("scikit_learn")).get(
+                "_estimator_type"
+            )
+
+            # Load based on model type
+            if model_type == "classifier":
+                model_object = xgb.XGBClassifier()
+                model_object.load_model(model_file)
+
+            else:
+                model_object = xgb.XGBRegressor()
+                model_object.load_model(model_file)
+
+            # Check type
+            if isinstance(model_object, XGBModel):
+                print(f"{model_file} is a model object.")
+
+                # Set return if type check passes
+                model_return = model_object
+            else:
+                print(f"{model_file} is NOT a model object.")
+
+        return model_return
+
+    @staticmethod
+    def load_from_joblib(tmpdir):
+        # Deprecation Warning
+
+        # Find the model file in the extracted directory
+        model_files = glob.glob(os.path.join(tmpdir, "*_model.joblib"))
+        if not model_files:
+            return None
+
+        # Instantiate return model
+        model_return = None
+
+        # Check each model_file for an XGBModel object
+        for model_file in model_files:
+            # Load the model
+            model_object = joblib.load(model_file)
+
+            # Check type
+            if isinstance(model_object, XGBModel):
+                print(f"{model_file} is a model object.")
+
+                # Set return if type check passes
+                model_return = model_object
+            else:
+                print(f"{model_file} is NOT a model object.")
+
+        # Return the model after exiting the temporary directory context
+        return model_return
+
+    def download_and_extract_model(self, model_artifact_uri):
```

Comment:
Yeah, this is a lot of new code, but shrug.. it's in utils and only used by Shap we'll let it go. Perhaps we can think about creating a helper method that loads xgboost models... double shrug...